lib/kino/control.ex

```diff
@@ -0,0 +1,137 @@
+defmodule Kino.Control do
+  @moduledoc """
+  Various widgets for user interactions.
+
+  Each widget is a UI control element that the user interacts
+  with, consequenty producing an event stream.
+
+  Those widgets are often useful paired with `Kino.Frame` for
+  presenting content that changes upon user interactions.
+
+  ## Examples
+
+  First, create a control and make sure it is rendered,
+  either by placing it at the end of a code cell or by
+  explicitly rendering it with `Kino.render/1`.
+
+      button = Kino.Control.button("Hello")
+
+  Next, to receive events from the control, a process needs to
+  subscribe to it and specify pick a name to distinguish the
+  events.
+
+      Kino.Control.subscribe(button, :hello)
+
+  As the user interacts with the button, the subscribed process
+  receives corresponding events.
+
+      IEx.Helpers.flush()
+      #=> {:event, :hello, %{origin: #PID<10895.9854.0>}}
+      #=> {:event, :hello, %{origin: #PID<10895.9854.0>}}
+  """
+
+  defstruct [:attrs]
+
+  @type t :: %__MODULE__{attrs: Kino.Output.control_attrs()}
+
+  defp new(attrs) do
+    ref = make_ref()
+    subscription_manager = Kino.SubscriptionManager.cross_node_name()
+
+    attrs = Map.merge(attrs, %{ref: ref, destination: subscription_manager})
+
+    Kino.Bridge.object_add_pointer(ref)
+
+    Kino.Bridge.object_add_release_hook(ref, fn ->
+      Kino.SubscriptionManager.clear_topic(ref)
+    end)
+
+    %__MODULE__{attrs: attrs}
+  end
+
+  @doc """
+  Creates a new button.
+  """
+  @spec button(String.t()) :: t()
+  def button(label) when is_binary(label) do
+    new(%{type: :button, label: label})
+  end
+
+  @doc """
+  Creates a new keyboard control.
+
+  This widget is represented as button that toggles interception
+  mode, in which the given keyboard events are captured.
+
+  ## Event info
+
+  In addition to standard properties, all events include additional
+  properties.
+
+  ### Key events
+
+    * `:type` - either `:keyup` or `:keydown`
+
+    * `:key` - the value matching the browser [KeyboardEvent.key](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key)
+
+  ### Status event
+
+    * `:type` - either `:status`
+
+    * `:enabled` - whether the keyboard is activated
```

Comment:
Not sure if it's better to have two separate events or a status event :thinking: